### ISSUE TITLE
[Telemetry] Share image

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#16076]
## Description & Product Manager / Data Scientist User Story
We are currently tracking all the actions listed in the Image Selection Action Menu (we call it context_menu.item_tapped) except for Share Image. 

As a product manager, I want to know how often users share an image to a third party app or another device, so I can measure whether improvements to this feature will help our users. 

## What questions will you answer with this data?
- How many users (and how often) share an image in a day?

## Acceptance Criteria
- [x] ENG filed a [DS JIRA](https://jira.mozilla.com/browse/DO-381) outlining the telemetry request.
- [ ] DS sign off on instrumentation methodology addressing product questions.
- [ ] Event pings can be queried via re:dash
- [ ] Event pings can be queried via amplitude
- [ ] We are sending telemetry events for the actions listed in the requirements
- [x] We have documented the telemetry
- [x] We have asked a data steward to [review](https://github.com/mozilla/data-review/blob/master/request.md) the telemetry

**Comment by Mugurell** [comment]
# Request for data collection review form

**All questions are mandatory. You must receive review from a data steward peer on your responses to these questions before shipping new data collection.**

_1) What questions will you answer with this data?_
How often users share an image from the app.

_2) Why does Mozilla need to answer these questions?  Are there benefits for users? Do we need this information to address product or business requirements?_
Measure whether improvements to this feature will help our users.

_3) What alternative methods did you consider to answer these questions? Why were they not sufficient?_
There is the only way. We are currently tracking all the actions listed in the Image Selection Action Menu (we call it context_menu.item_tapped) except for Share Image.

_4) Can current instrumentation answer these questions?_
No. We need a new specific event for a specific app feature.

_5) List all proposed measurements and indicate the category of data collection for each measurement, using the [Firefox data collection categories](https://wiki.mozilla.org/Firefox/Data_Collection) found on the Mozilla wiki._

_**Note that the data steward reviewing your request will characterize your data collection based on the highest (and most sensitive) category.**_

<table>
  <tr>
    <td>Measurement Description</td>
    <td>Data Collection Category</td>
    <td>Tracking Bug #</td>
  </tr>
  <tr>
    <td>Share image events</td>
    <td>Category 2</td>
    <td>https://github.com/mozilla-mobile/fenix/issues/16076</td>
  </tr>
</table>

_6) Please provide a link to the documentation for this data collection which describes the ultimate data set in a public, complete, and accurate way._
This collection is documented in the `metrics.md` document document included in the project - https://github.com/mozilla-mobile/fenix/blob/master/docs/metrics.md

_7) How long will this data be collected?  Choose one of the following:_
Based on https://github.com/mozilla-mobile/fenix/pull/13958#issuecomment-676857877 until 2021-04-01.
Pending renewal.

_8) What populations will you measure?_
All release channels and locales.

_9) If this data collection is default on, what is the opt-out mechanism for users?_
 Users can opt of of data collection by disabling Usage and technical data from Settings -> Data collection.

_10) Please provide a general description of how you will analyze this data._
 Glean / Amplitude

_11) Where do you intend to share the results of your analysis?_
 Only on Glean, Amplitude and with mobile teams.

_12) Is there a third-party tool (i.e. not Telemetry) that you are proposing to use for this data collection?_
No.

**Comment by Mugurell** [comment]
@boek Can you help with a data review?

**Comment by boek** [comment]
data-review r+

# Data Review Form (to be filled by Data Stewards)

Instructions: Data Stewards will review a request for data collection and endorse responses to each question. **If the request does not provide answers to questions, reviewers give an r- and point to the questions that can’t be answered.**

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way?
Yes, metrics.md and metrics.yaml

2) Is there a control mechanism that allows the user to turn the data collection on and off?
Yes data controls in fenix settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?
Modifies existing metric, expires "2021-04-01". Fenix team will monitor

4) Using the **[category system of data types](https://wiki.mozilla.org/Firefox/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?
Type 2

5) Is the data collection request for default-on or default-off?
Default on

6) Does the instrumentation include the addition of **any *new* identifiers** (whether anonymous or otherwise; e.g., username, random IDs, etc.  See the appendix for more details)?
No

7) Is the data collection covered by the existing Firefox privacy notice? **If unsure: escalate to legal if:**

* The data includes new identifiers; OR

* The data falls within the Web activity category AND is default-on.

8) Does there need to be a check-in in the future to determine whether to renew the data? (Yes/No) (If yes, set a todo reminder or file a bug if appropriate)**

9) Does the data collection use a third-party collection tool? **If yes, escalate to legal.**

**Comment by Mugurell** [comment]
With this being already merged I think it's ready to be tested.

**Comment by AndiAJ** [comment]
Hi, **verified as fixed** on both the latest **`Nightly 2011192`** and **`Beta 84.0.0-beta.1`** using a Google Pixel 3a (Android 11)

► Nightly 201119
✔️ Share image - context menu from a link - Ping **b98123a5-9520-41a3-894d-3a8914bf346c**
✔️ Share image - context menu from the picture itself - Ping **f5e5567b-be7b-4e06-aea6-405fa655d112**

[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-t1911000)
[Logcat](https://github.com/mozilla-mobile/fenix/files/5567125/imgn.txt)


► Beta 84.0.0-beta.1
✔️ Share image - context menu from a link - Ping **a87e0464-8716-48c1-baa9-d98d559cf956**
✔️ Share image - context menu from the picture itself - Ping **93e643d1-f3ec-414a-ad55-e87178cdf153**

[Glean dashboard](https://debug-ping-preview.firebaseapp.com/pings/aj-t1911003)
[Logcat](https://github.com/mozilla-mobile/fenix/files/5567126/imgb.txt)

```
{
          "category": "context_menu",
          "extra": {
            "named": "share_image"
          },
          "name": "item_tapped",
          "timestamp": 8933
        }
```
